pkg/util/fileIndexer.go

```diff
@@ -0,0 +1,165 @@
+package util
+
+import (
+	"encoding/json"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const fileIndexDirectory = ".odo"
+const fileIndexName = "odo-file-index.json"
+
+type FileData struct {
+	Size             int64
+	LastModifiedDate time.Time
+}
+
+// read tries to read the odo index file from the given location and returns the data from the file
+// if no such file is present, it means the folder hasn't been walked and thus returns a empty list
+func read(filePath string) (map[string]FileData, error) {
+	// read operation
+	fileReadMap := make(map[string]FileData)
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		return map[string]FileData{}, nil
+	}
+
+	jsonFileRead, err := os.Open(filePath)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFileRead.Close()
+
+	byteValue, _ := ioutil.ReadAll(jsonFileRead)
```

Comment:
Comments here would be great, this reads in the bytes, unmarshals and then... what?